File: 2021/Day02/Dive.java
import java.util.*;
import java.io.File;

public class Dive {
    public static void main(String[] args){
        try {
            Scanner sc = new Scanner(new File("input.txt"));
            int horizon = 0;
            int depth = 0;
            while(sc.hasNext()){
                String[] nextLine = sc.nextLine().split(" ");
                switch (nextLine[0]) {
                    case "forward":
                        horizon += Integer.parseInt(nextLine[1]);
                        break;
                    case "down":
                        depth += Integer.parseInt(nextLine[1]);
                        break;
                    case "up":
                        depth -= Integer.parseInt(nextLine[1]);
                        break;
                    default:
                        break;
                }
            }
            System.out.println(horizon*depth);
        } catch (Exception e) {
            //TODO: handle exception
        }
    }
}

File: 2021/python/Day02/Dive.py
file = open('input.txt', 'r')
horizson = 0
depth = 0
for line in file:
    nextLine = line.split()
    if(nextLine[0] == "forward"):
        horizson += int(nextLine[1])
    elif(nextLine[0] == "down"):
        depth += int(nextLine[1])
    else:
        depth -= int(nextLine[1])
print(horizson*depth)
